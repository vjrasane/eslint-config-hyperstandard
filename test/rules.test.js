import { CLIEngine } from 'eslint'
import { join, dirname } from 'path'

it('validate rules', () => {
  const engine = new CLIEngine({
    useEslintrc: false,
    configFile: join(dirname(module.filename), '../src/.hyperstandard.eslintrc.js')
  })

  const rules = [ ...engine.linter.rules.getAllLoadedRules().keys() ]
  expect(rules).toMatchSnapshot()
})
