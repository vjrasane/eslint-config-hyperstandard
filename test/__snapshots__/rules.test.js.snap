// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate rules 1`] = `
Array [
  "accessor-pairs",
  "array-bracket-newline",
  "array-bracket-spacing",
  "array-callback-return",
  "array-element-newline",
  "arrow-body-style",
  "arrow-parens",
  "arrow-spacing",
  "block-scoped-var",
  "block-spacing",
  "brace-style",
  "callback-return",
  "camelcase",
  "capitalized-comments",
  "class-methods-use-this",
  "comma-dangle",
  "comma-spacing",
  "comma-style",
  "complexity",
  "computed-property-spacing",
  "consistent-return",
  "consistent-this",
  "constructor-super",
  "curly",
  "default-case",
  "dot-location",
  "dot-notation",
  "eol-last",
  "eqeqeq",
  "for-direction",
  "func-call-spacing",
  "func-name-matching",
  "func-names",
  "func-style",
  "function-paren-newline",
  "generator-star-spacing",
  "getter-return",
  "global-require",
  "guard-for-in",
  "handle-callback-err",
  "id-blacklist",
  "id-length",
  "id-match",
  "implicit-arrow-linebreak",
  "indent-legacy",
  "indent",
  "init-declarations",
  "jsx-quotes",
  "key-spacing",
  "keyword-spacing",
  "line-comment-position",
  "linebreak-style",
  "lines-around-comment",
  "lines-around-directive",
  "lines-between-class-members",
  "max-classes-per-file",
  "max-depth",
  "max-len",
  "max-lines-per-function",
  "max-lines",
  "max-nested-callbacks",
  "max-params",
  "max-statements-per-line",
  "max-statements",
  "multiline-comment-style",
  "multiline-ternary",
  "new-cap",
  "new-parens",
  "newline-after-var",
  "newline-before-return",
  "newline-per-chained-call",
  "no-alert",
  "no-array-constructor",
  "no-async-promise-executor",
  "no-await-in-loop",
  "no-bitwise",
  "no-buffer-constructor",
  "no-caller",
  "no-case-declarations",
  "no-catch-shadow",
  "no-class-assign",
  "no-compare-neg-zero",
  "no-cond-assign",
  "no-confusing-arrow",
  "no-console",
  "no-const-assign",
  "no-constant-condition",
  "no-continue",
  "no-control-regex",
  "no-debugger",
  "no-delete-var",
  "no-div-regex",
  "no-dupe-args",
  "no-dupe-class-members",
  "no-dupe-keys",
  "no-duplicate-case",
  "no-duplicate-imports",
  "no-else-return",
  "no-empty-character-class",
  "no-empty-function",
  "no-empty-pattern",
  "no-empty",
  "no-eq-null",
  "no-eval",
  "no-ex-assign",
  "no-extend-native",
  "no-extra-bind",
  "no-extra-boolean-cast",
  "no-extra-label",
  "no-extra-parens",
  "no-extra-semi",
  "no-fallthrough",
  "no-floating-decimal",
  "no-func-assign",
  "no-global-assign",
  "no-implicit-coercion",
  "no-implicit-globals",
  "no-implied-eval",
  "no-inline-comments",
  "no-inner-declarations",
  "no-invalid-regexp",
  "no-invalid-this",
  "no-irregular-whitespace",
  "no-iterator",
  "no-label-var",
  "no-labels",
  "no-lone-blocks",
  "no-lonely-if",
  "no-loop-func",
  "no-magic-numbers",
  "no-misleading-character-class",
  "no-mixed-operators",
  "no-mixed-requires",
  "no-mixed-spaces-and-tabs",
  "no-multi-assign",
  "no-multi-spaces",
  "no-multi-str",
  "no-multiple-empty-lines",
  "no-native-reassign",
  "no-negated-condition",
  "no-negated-in-lhs",
  "no-nested-ternary",
  "no-new-func",
  "no-new-object",
  "no-new-require",
  "no-new-symbol",
  "no-new-wrappers",
  "no-new",
  "no-obj-calls",
  "no-octal-escape",
  "no-octal",
  "no-param-reassign",
  "no-path-concat",
  "no-plusplus",
  "no-process-env",
  "no-process-exit",
  "no-proto",
  "no-prototype-builtins",
  "no-redeclare",
  "no-regex-spaces",
  "no-restricted-globals",
  "no-restricted-imports",
  "no-restricted-modules",
  "no-restricted-properties",
  "no-restricted-syntax",
  "no-return-assign",
  "no-return-await",
  "no-script-url",
  "no-self-assign",
  "no-self-compare",
  "no-sequences",
  "no-shadow-restricted-names",
  "no-shadow",
  "no-spaced-func",
  "no-sparse-arrays",
  "no-sync",
  "no-tabs",
  "no-template-curly-in-string",
  "no-ternary",
  "no-this-before-super",
  "no-throw-literal",
  "no-trailing-spaces",
  "no-undef-init",
  "no-undef",
  "no-undefined",
  "no-underscore-dangle",
  "no-unexpected-multiline",
  "no-unmodified-loop-condition",
  "no-unneeded-ternary",
  "no-unreachable",
  "no-unsafe-finally",
  "no-unsafe-negation",
  "no-unused-expressions",
  "no-unused-labels",
  "no-unused-vars",
  "no-use-before-define",
  "no-useless-call",
  "no-useless-computed-key",
  "no-useless-concat",
  "no-useless-constructor",
  "no-useless-escape",
  "no-useless-rename",
  "no-useless-return",
  "no-var",
  "no-void",
  "no-warning-comments",
  "no-whitespace-before-property",
  "no-with",
  "nonblock-statement-body-position",
  "object-curly-newline",
  "object-curly-spacing",
  "object-property-newline",
  "object-shorthand",
  "one-var-declaration-per-line",
  "one-var",
  "operator-assignment",
  "operator-linebreak",
  "padded-blocks",
  "padding-line-between-statements",
  "prefer-arrow-callback",
  "prefer-const",
  "prefer-destructuring",
  "prefer-numeric-literals",
  "prefer-object-spread",
  "prefer-promise-reject-errors",
  "prefer-reflect",
  "prefer-rest-params",
  "prefer-spread",
  "prefer-template",
  "quote-props",
  "quotes",
  "radix",
  "require-atomic-updates",
  "require-await",
  "require-jsdoc",
  "require-unicode-regexp",
  "require-yield",
  "rest-spread-spacing",
  "semi-spacing",
  "semi-style",
  "semi",
  "sort-imports",
  "sort-keys",
  "sort-vars",
  "space-before-blocks",
  "space-before-function-paren",
  "space-in-parens",
  "space-infix-ops",
  "space-unary-ops",
  "spaced-comment",
  "strict",
  "switch-colon-spacing",
  "symbol-description",
  "template-curly-spacing",
  "template-tag-spacing",
  "unicode-bom",
  "use-isnan",
  "valid-jsdoc",
  "valid-typeof",
  "vars-on-top",
  "wrap-iife",
  "wrap-regex",
  "yield-star-spacing",
  "yoda",
  "import/no-unresolved",
  "import/named",
  "import/default",
  "import/namespace",
  "import/no-namespace",
  "import/export",
  "import/no-mutable-exports",
  "import/extensions",
  "import/no-restricted-paths",
  "import/no-internal-modules",
  "import/group-exports",
  "import/no-relative-parent-imports",
  "import/no-self-import",
  "import/no-cycle",
  "import/no-named-default",
  "import/no-named-as-default",
  "import/no-named-as-default-member",
  "import/no-anonymous-default-export",
  "import/no-commonjs",
  "import/no-amd",
  "import/no-duplicates",
  "import/first",
  "import/max-dependencies",
  "import/no-extraneous-dependencies",
  "import/no-absolute-path",
  "import/no-nodejs-modules",
  "import/no-webpack-loader-syntax",
  "import/order",
  "import/newline-after-import",
  "import/prefer-default-export",
  "import/no-default-export",
  "import/no-dynamic-require",
  "import/unambiguous",
  "import/no-unassigned-import",
  "import/no-useless-path-segments",
  "import/dynamic-import-chunkname",
  "import/exports-last",
  "import/no-deprecated",
  "import/imports-first",
  "node/exports-style",
  "node/no-deprecated-api",
  "node/no-extraneous-import",
  "node/no-extraneous-require",
  "node/no-missing-import",
  "node/no-missing-require",
  "node/no-unpublished-bin",
  "node/no-unpublished-import",
  "node/no-unpublished-require",
  "node/no-unsupported-features/es-builtins",
  "node/no-unsupported-features/es-syntax",
  "node/no-unsupported-features/node-builtins",
  "node/prefer-global/buffer",
  "node/prefer-global/console",
  "node/prefer-global/process",
  "node/prefer-global/url-search-params",
  "node/prefer-global/url",
  "node/process-exit-as-throw",
  "node/shebang",
  "node/no-hide-core-modules",
  "node/no-unsupported-features",
  "promise/param-names",
  "promise/no-return-wrap",
  "promise/always-return",
  "promise/catch-or-return",
  "promise/prefer-await-to-callbacks",
  "promise/prefer-await-to-then",
  "promise/no-native",
  "promise/no-callback-in-promise",
  "promise/no-promise-in-callback",
  "promise/no-nesting",
  "promise/avoid-new",
  "promise/no-new-statics",
  "promise/no-return-in-finally",
  "promise/valid-params",
  "standard/array-bracket-even-spacing",
  "standard/computed-property-even-spacing",
  "standard/object-curly-even-spacing",
  "standard/no-callback-literal",
]
`;
